### PR TITLE
Integrate Rockset as Vectorstore

### PR DESCRIPTION
**Pull request by anubhav94N**
#### Description

This PR adds Rockset as a vectorstore for langchain. [Rockset](https://rockset.com/blog/introducing-vector-search-on-rockset/) is a real time OLAP database which provides a fast and efficient vector search functionality. Further since it is entirely schemaless, it can store metadata in separate columns thereby allowing fast metadata filters during vector similarity search (as opposed to storing the entire metadata in a single JSON column). It currently supports three distance functions: `COSINE_SIMILARITY`, `EUCLIDEAN_DISTANCE`, and `DOT_PRODUCT`. 

This PR adds `rockset` client as an optional dependency. 

We would love a twitter shoutout, our handle is https://twitter.com/RocksetCloud

#### Before submitting

1. Integration test: https://github.com/anubhav94N/langchain/blob/master/tests/integration_tests/vectorstores/test_rocksetdb.py
2. Example notebook: https://github.com/anubhav94N/langchain/blob/master/docs/modules/indexes/vectorstores/examples/rockset_vector_database.ipynb
3. Ran `make format` and `make lint` locally

#### Who can review?

@hwchase17, @dev2049 can you help review please?

### CODE REVIEW

File: langchain/vectorstores/rocksetdb.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+            text_key: column in Rockset collection to use to store the text
+            embedding_key: column in Rockset collection to use to store the embedding.
+                           Note: We must apply `VECTOR_ENFORCE()` on this column via
+                           Rockset ingest transformation.
+
+        """
+        try:
+            from rockset import RocksetClient
+        except ImportError:
+            raise ValueError(
```
Comment (leo-gan) [inline langchain/vectorstores/rocksetdb.py:69]: should be
`raise ImportError(`

Comment (anubhav94N) [reply langchain/vectorstores/rocksetdb.py]: Updated - thanks!

**Review by hwchase17 — APPROVED** [review]
some merge conflicts, otherwise lgtm

**Comment by dev2049** [comment]
thanks @anubhav94N!

**Pull request merged**